addons/mail/static/src/discuss/call/common/ptt_extension_service.js

```diff
@@ -0,0 +1,68 @@
+/* @odoo-module */
+
+import { browser } from "@web/core/browser/browser";
+import { registry } from "@web/core/registry";
+
+export const pttExtensionHookService = {
+    start(env) {
+        const INITIAL_RELEASE_TIMEOUT = 500;
+        const COMMON_RELEASE_TIMEOUT = 200;
+        let isEnabled = false;
+        let voiceActivated = false;
+
+        browser.addEventListener("message", ({ data }) => {
+            const rtc = env.services["discuss.rtc"];
```

Comment:
Shouldn't you add discuss.rtc as a dependency of your service and retrieve it from the params of start?

Reply:
This would lead to circular dependencies. This service abstracts the interaction with the extension but acts like an extension of the rtc service.